torchao/dtypes/nf4tensor.py

```diff
@@ -428,7 +462,7 @@ def quantize_tensor_nearest(
     # pyre-fixme[40]: Static method `dequantize` cannot override a non-static method
     #  defined in `torch._C.TensorBase`.
     def dequantize(value: torch.Tensor, nf4: torch.Tensor) -> torch.Tensor:
-        """Dequantize a nf4 value to float16 format"""
+        """Dequantize a nf4 value to bfloat16 format"""
```

Comment:
Is nf4 tensor still restricted to bf16 only for the higher precision, are there any blockers in supporting fp32?

Reply:
We should be able to support arbitrary precision for conversion, but of course the fidelity of nf4 is independen of the dtype that was passed during construction.